mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -592,6 +629,59 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
   }
 
   /**
+    * Tweedie exponential family distribution.
+    * The default link for the Tweedie family is the log link.
+    */
+  private[regression] object Tweedie extends Family("tweedie") {
+
+    val defaultLink: Link = Log
+
+    var variancePower: Double = 1.5
+
+    override def initialize(y: Double, weight: Double): Double = {
+      if (variancePower > 1.0 && variancePower < 2.0) {
+        require(y >= 0.0, "The response variable of the specified Tweedie distribution " +
+          s"should be non-negative, but got $y")
+        math.max(y, 0.1)
```

Comment:
I have not seen a formal justification for the choice of 0.1 in R. This seminal [paper](http://users.du.se/~lrn/StatMod10/HomeExercise2/Nelder_Pregibon.pdf) suggests 1/6 (about 0.17) to be the best constant. I would prefer to be consistent with R so that we can make comparison. Using a constant is a good idea.